services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy

```diff
@@ -0,0 +1,81 @@
+package org.openkilda.functionaltests.spec.configuration
+
+import static org.openkilda.functionaltests.extension.tags.Tag.LOW_PRIORITY
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.messaging.model.system.KildaConfigurationDto
+import org.openkilda.model.FlowEncapsulationType
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Tags(LOW_PRIORITY)
+@Narrative("""Kilda configuration is a special lever that allows to default flow encapsulation type while creating.
+This spec assumes that 'transit_vlan' is always default type""")
+class ConfigurationSpec extends HealthCheckSpecification {
+    @Shared
+    FlowEncapsulationType defaultEncapsulationType = FlowEncapsulationType.TRANSIT_VLAN
+
+    def "System allows to get/update kilda configuration"() {
+        when: "Get kilda configuration"
+        def getResponse = northbound.getKildaConfiguration()
+
+        then: "Kilda configuration is returned"
+        getResponse.flowEncapsulationType == defaultEncapsulationType.toString().toLowerCase()
+
+        when: "Update kilda configuration"
+        def newFlowEncapsulationType = FlowEncapsulationType.VXLAN
+        def updateResponse = northbound.updateKildaConfiguration(
+                new KildaConfigurationDto(flowEncapsulationType: newFlowEncapsulationType))
+
+        then: "Correct response is returned"
+        updateResponse.flowEncapsulationType == newFlowEncapsulationType.toString().toLowerCase()
+
+        and: "Kilda configuration is really updated"
+        northbound.getKildaConfiguration().flowEncapsulationType == newFlowEncapsulationType.toString().toLowerCase()
+
+        cleanup: "Restore default configuration"
+        northbound.updateKildaConfiguration(
+                new KildaConfigurationDto(flowEncapsulationType: defaultEncapsulationType))
+    }
+
+    def "System takes into account default flow encapsulation type while creating flow"() {
+        given: "Current default flow encapsulation type"
+        def currentConfiguration = northbound.getKildaConfiguration()
+
+        when: "Create a flow without encapsulation type"
+        def switchPair = topologyHelper.getRandomSwitchPair()
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.encapsulationType = null
+        northbound.addFlow(flow)
+
+        then: "Flow is created with correct encapsulation type"
+        northbound.getFlow(flow.id).encapsulationType == currentConfiguration.flowEncapsulationType
+
+        cleanup: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
```

Comment:
The spec is obviously missing a test case that would update the default encap type and check that flow creates with an updated new value. The above test may be just a coincidence. How do you feel about merging 1st and 2nd tests into one and checking config update together with creating a flow?

Reply:
I had second thoughts about missing test case while writing this spec. 
but now I see it have to be added. 
So: 
- missing test case was added with `HARDWARE` tag.
- 1st and 2nd tests were merged.
- this test covers v2 only

Reply:
I don't like this test. How do you know that system is not _always_ creating a vxlan flow when encap type is missing? As for me the steps should be as follows
```
1. Create a flow without encap
2. Check that flow has transit_vlan (current default)
3. update current default encap type to vxlan
4. create one more flow
5. Check that the encap type has also changed to vxlan
```

Reply:
I can't argue with that (I tried to save time for running this test)

Fixed.